frontend/src/component/common/ConstraintAccordion/ConstraintAccordionEdit/ConstraintAccordionEditBody/DateSingleValue/DateSingleValue.tsx

```diff
@@ -1,41 +1,122 @@
+        if (pickedDate && timezone) {
+            const utc = zonedTimeToUtc(pickedDate!, timezone);
+            setValue(utc.toISOString());
+            const localDate = utcToZonedTime(utc, localTimezoneName);
+            setLocalDate(localDate.toISOString());
+        }
+    }, [localTimezoneName, pickedDate, setValue, timezone]);
+
     if (!value) return null;
 
     return (
         <>
             <ConstraintFormHeader>Select a date</ConstraintFormHeader>
-            <Input
-                id="date"
-                label="Date"
-                type="datetime-local"
-                value={parseDateValue(value)}
-                onChange={e => {
-                    setError('');
-                    const parsedDate = parseValidDate(e.target.value);
-                    const dateString = parsedDate?.toISOString();
-                    dateString && setValue(dateString);
-                }}
-                InputLabelProps={{
-                    shrink: true,
-                }}
-                error={Boolean(error)}
-                errorText={error}
-                required
+            <StyledWrapper>
+                <Input
+                    id="date"
+                    label="Date"
+                    type="datetime-local"
+                    value={parseDateValue(pickedDate)}
+                    onChange={e => {
+                        setError('');
+                        const parsedDate = parseValidDate(e.target.value);
+                        const dateString = parsedDate?.toISOString();
+                        dateString && setPickedDate(dateString);
+                    }}
+                    InputLabelProps={{
+                        shrink: true,
+                    }}
+                    error={Boolean(error)}
+                    errorText={error}
+                    required
+                />
+                <ConditionallyRender
+                    condition={showSelectableTimezone}
+                    show={
+                        <ConditionallyRender
+                            condition={timezones.length > 0}
+                            show={
+                                <GeneralSelect
+                                    options={timezones}
+                                    label={'Timezone'}
+                                    onChange={setTimezone}
+                                    value={timezone}
+                                    sx={{ minWidth: 180 }}
+                                />
+                            }
+                        />
+                    }
+                    elseShow={
+                        <ConditionallyRender
+                            condition={Boolean(localTimezone)}
+                            show={
+                                <p>
+                                    {localTimezone?.key}{' '}
+                                    {`(UTC ${localTimezone?.utcOffset})`}
+                                </p>
```

Comment:
What do we render if localTimezone is undefined? Is this just an empty paragraph? Can we render something sensible here in that case?

Reply:
How about 'No timezone information available'

Reply:
Sure, that's an option, but it doesn't tell the user what time zone the time we're showing them is? I don't know exactly when this will be undefined, but what about something like:

> The time shown is in your local time zone according to your browser.

It seems that the only times we won't have it, is when we can't find the key in the list of time zones, yeah?

Reply:
yes